plugins/kubernetes/app/controllers/kubernetes_dashboard_controller.rb

```diff
@@ -0,0 +1,82 @@
+require 'kubeclient'
+
+class KubernetesDashboardController < ApplicationController
+  include CurrentProject
+
+  def index
+    render json: format_pod_info(pod_info).to_json
+  end
+
+  private
+
+  def pod_info
+    roles = {}
+    Environment.find(params[:environment]).cluster_deploy_groups.each do |cluster_deploy_group|
+      cluster_deploy_group.cluster.client.get_pods(
+          namespace: cluster_deploy_group.namespace,
+          label_selector: "project=#{current_project.name_for_label}").each do |pod|
+        labels = pod.metadata.labels
+        roles[labels.role_id] ||= build_role(labels.role_id)
+        role = roles[labels.role_id]
+        deploy_group_id = cluster_deploy_group.deploy_group.id
+        role[:deploy_groups][deploy_group_id] ||= build_deploy_group(deploy_group_id)
+        deploy_group = role[:deploy_groups][deploy_group_id]
+        deploy_group[:releases][labels.release_id] ||= build_release(labels.release_id, labels.role_id)
+        release = deploy_group[:releases][labels.release_id]
+        api_pod = Kubernetes::Api::Pod.new(pod)
+        release[:live_replicas] += 1 if api_pod.ready?
+      end
+    end
+    roles
+  end
```

Comment:
Just a suggestion: could this logic be refactored into a separate method as well, as you did for all the methods below? I mean, something in the likes of:
```
def pod_info
  Environment.find(params[:environment]).cluster_deploy_groups.each do |cluster_deploy_group|
    cluster_deploy_group.cluster.client.get_pods(
      namespace: cluster_deploy_group.namespace,
      label_selector: "project=#{current_project.name_for_label}").
      each(&:extract_pod_info)
  end
  roles
end

def extract_pod_info
   ... #all the logic responsible for parsing the data from the pod as it stands
end

def roles
  @roles ||= {}
end
```

Just to extract the parsing of the data from the actual call.